h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -84,7 +84,7 @@ private void imputeMissingWithMeans(double[] data) {
 
     if (data.length == nfeatures()) { // using centered gam cols, nfeatures denotes centered gam columns
       for (int ind = 0; ind < _numsCenter; ind++)
-        if (Double.isNaN(data[ind + _cats])) data[ind] = _numNAFillsCenter[ind];
+        if (Double.isNaN(data[ind + _cats])) data[ind + _cats] = _numNAFillsCenter[ind];
```

Comment:
Please note this part has wrong indexing and is being fixed: https://github.com/h2oai/h2o-3/pull/4832 (but not merged yet at the moment).

We might copy & paste the code from there if we want to.

Reply:
I would like to keep this here because it is a bug and it makes me sad to ignore it.